Native compiled apps do not start with Vaadin 25 beta1

### Description of the bug

```
npm init vaadin@latest v25-native
cd v25-native
mvn -Pproduction -Pnative native:compile
target/v25-native
```

results in
```
Caused by: java.lang.RuntimeException: no DevModeHandlerManager implementation found but but dev server enabled. Either disable by setting vaadin.frontend.hotdeploy=false (and run the build-frontend maven goal) or include the vaadin-dev-server dependency
	at com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener.failFastContextInitialized(VaadinServletContextInitializer.java:528) ~[na:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$FailFastServletContextListener.contextInitialized(VaadinServletContextInitializer.java:202) ~[v25-native:na]
```

### Expected behavior

It should start in production mode

### Minimal reproducible example

as above

### Versions

- Vaadin / Flow version:
- Java version:
- OS version:
- Browser version (if applicable):
- Application Server (if applicable):
- IDE (if applicable):

Tested again with the linked path, and now at runtime I get errors related to Hibernate, like the following

```
Caused by: java.lang.NoSuchMethodException: org.hibernate.boot.model.relational.ColumnOrderingStrategyStandard.<init>()
        at java.base@23.0.1/java.lang.Class.checkConstructor(DynamicHub.java:1171)
        at java.base@23.0.1/java.lang.Class.getConstructor0(DynamicHub.java:1365)
        at java.base@23.0.1/java.lang.Class.getDeclaredConstructor(DynamicHub.java:3004)
        at org.hibernate.boot.registry.selector.internal.StrategySelectorImpl.create(StrategySelectorImpl.java:178)
        ... 34 more
```

```
Caused by: java.lang.IllegalArgumentException: Invalid logger interface org.hibernate.boot.archive.scan.internal.ScannerLogger (implementation not found)
        at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2630)
        at org.jboss.logging.Logger.getMessageLogger(Logger.java:2536)
        at org.jboss.logging.Logger.getMessageLogger(Logger.java:2521)
        at org.hibernate.boot.archive.scan.internal.ScannerLogger.<clinit>(ScannerLogger.java:33)
```


To be noted that the above errors happen only with Vaadin 25. With Vaadin 24 the native executable works fine

--- a/vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinHintsRegistrar.java
+++ b/vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinHintsRegistrar.java
@@ -50,7 +50,7 @@ public class VaadinHintsRegistrar implements RuntimeHintsRegistrar {
                 .forEach(cls -> ref.registerType(cls, MemberCategory.values()));
 
         // Bundles, build info etc
-        hints.resources().registerPattern("META-INF/VAADIN/*");
+        hints.resources().registerPattern("META-INF/VAADIN/**");
         hints.resources().registerPattern("vaadin-i18n/*");
         hints.resources().registerPattern("vaadin-featureflags.properties");
 
@@ -71,8 +71,8 @@ public class VaadinHintsRegistrar implements RuntimeHintsRegistrar {
 
         // Flow server resources like BootstrapHandler.js and
         // RouteNotFoundError_prod.html
-        hints.resources().registerPattern("com/vaadin/flow/server/*");
-        hints.resources().registerPattern("com/vaadin/flow/router/*");
+        hints.resources().registerPattern("com/vaadin/flow/server/**");
+        hints.resources().registerPattern("com/vaadin/flow/router/**");
     }
 
     private void registerResourceIfPresent(RuntimeHints hints, String path) {

--- a/vaadin-spring/src/test/java/com/vaadin/flow/spring/springnative/VaadinHintsRegistrarTest.java
+++ b/vaadin-spring/src/test/java/com/vaadin/flow/spring/springnative/VaadinHintsRegistrarTest.java
@@ -1,0 +1,77 @@
+/*
+ * Copyright 2000-2025 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.spring.springnative;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.predicate.RuntimeHintsPredicates;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class VaadinHintsRegistrarTest {
+
+    @Test
+    void shouldRegisterVaadinMetadata() {
+        RuntimeHints hints = new RuntimeHints();
+        new VaadinHintsRegistrar().registerHints(hints,
+                getClass().getClassLoader());
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("META-INF/VAADIN/conf/flow-build-info.json"))
+                .accepts(hints);
+    }
+
+    @Test
+    void shouldRegisterFeatureFlags() {
+        RuntimeHints hints = new RuntimeHints();
+        new VaadinHintsRegistrar().registerHints(hints,
+                getClass().getClassLoader());
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("vaadin-featureflags.properties")).accepts(hints);
+    }
+
+    @Test
+    void shouldRegisterDefaultI18NPropertiesFiles() {
+        RuntimeHints hints = new RuntimeHints();
+        new VaadinHintsRegistrar().registerHints(hints,
+                getClass().getClassLoader());
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("vaadin-i18n/translations.properties"))
+                .accepts(hints);
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("vaadin-i18n/translations_en.properties"))
+                .accepts(hints);
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("vaadin-i18n/translations_it_IT.properties"))
+                .accepts(hints);
+    }
+
+    @Test
+    void shouldRegisterFlowServerStaticResources() {
+        RuntimeHints hints = new RuntimeHints();
+        new VaadinHintsRegistrar().registerHints(hints,
+                getClass().getClassLoader());
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("com/vaadin/flow/router/NoRoutesError_dev.html"))
+                .accepts(hints);
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("com/vaadin/flow/server/BootstrapHandler.js"))
+                .accepts(hints);
+        assertThat(RuntimeHintsPredicates.resource()
+                .forResource("com/vaadin/flow/server/frontend/index.ts"))
+                .accepts(hints);
+    }
+
+}

